tools/testing/selftests/kvm/priv_memfd_test.c

```diff
@@ -902,24 +1033,65 @@ static void setup_and_execute_test(uint32_t test_id)
 	priv_memfd_testsuite[test_id].priv_memfd = priv_memfd;
 	vcpu_work(vm, test_id);
 
-	munmap(shared_mem, TEST_MEM_SIZE);
+	munmap(shared_mem, mem_size);
 	priv_memfd_testsuite[test_id].shared_mem = NULL;
 	close(priv_memfd);
 	priv_memfd_testsuite[test_id].priv_memfd = -1;
 	kvm_vm_free(vm);
 }
 
+bool is_using_hugepages(const struct page_combo pages)
+{
+	return !(pages.shared == PAGE_4KB && pages.private == PAGE_4KB);
+}
+
+void hugepage_requirements_text(const struct page_combo matrix)
+{
+	int _2mb_pages_needed = 0;
+	int _1gb_pages_needed = 0;
+	enum page_size sizes[] = { matrix.shared, matrix.private };
+
+	for (int i = 0; i < ARRAY_SIZE(sizes); ++i) {
+		if (sizes[i] == PAGE_2MB)
+			++_2mb_pages_needed;
+		if (sizes[i] == PAGE_1GB)
+			++_1gb_pages_needed;
+	}
+	if (_2mb_pages_needed != 0 && _1gb_pages_needed != 0) {
+		pr_info("This test requires %d 2MB page(s) and %d 1GB page(s)\n",
+				_2mb_pages_needed, _1gb_pages_needed);
+	} else if (_2mb_pages_needed != 0) {
+		pr_info("This test requires %d 2MB page(s)\n", _2mb_pages_needed);
+	} else if (_1gb_pages_needed != 0) {
+		pr_info("This test requires %d 1GB page(s)\n", _1gb_pages_needed);
+	}
+}
+
 int main(int argc, char *argv[])
 {
 	/* Tell stdout not to buffer its content */
 	setbuf(stdout, NULL);
 
+	struct page_combo page_size_matrix[] = {
```

Comment:
@adiviness : Lets allow an option to execute hugepage specific tests conditionally.